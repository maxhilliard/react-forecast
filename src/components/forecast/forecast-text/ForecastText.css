.forecastText {
    padding: 20px;
}