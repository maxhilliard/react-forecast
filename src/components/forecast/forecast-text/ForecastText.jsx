import React from 'react';
import PropTypes from 'prop-types';

import styles from './ForecastText.css';

const TEXT_MAP = {
    INSTRUCTIONS: 'Search for a city',
    ERROR: '(◡﹏◡) There was an error with your search',
};

const ForecastText = ({ isForecastError }) => (
    <div className={styles.forecastText}>
        <h2>
            {isForecastError ? TEXT_MAP.ERROR : TEXT_MAP.INSTRUCTIONS}
        </h2>
    </div>
);

ForecastText.propTypes = {
    isForecastError: PropTypes.bool,
};

ForecastText.defaultProps = {
    isForecastError: false,
};

export default ForecastText;
