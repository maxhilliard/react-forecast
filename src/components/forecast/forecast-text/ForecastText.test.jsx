import React from 'react';
import { shallow } from 'enzyme';

import ForecastText from './ForecastText';

describe('ForecastText', () => {
    const wrapper = shallow(<ForecastText />);

    describe('snapshot', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('props', () => {
        it('should render the expected text given isForecastError prop is falsy', () => {
            const expectedText = 'Search for a city';
            const actualText = wrapper.find('h2').text();

            expect(actualText).toEqual(expectedText);
        });

        it('should render the expected text given isForecastError prop is truthy', () => {
            wrapper.setProps({
                isForecastError: true,
            });

            const expectedText = '(◡﹏◡) There was an error with your search';
            const actualText = wrapper.find('h2').text();

            expect(actualText).toEqual(expectedText);
        });
    });
});
