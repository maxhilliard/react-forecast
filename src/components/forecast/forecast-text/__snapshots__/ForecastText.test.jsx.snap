// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastText snapshot should match snapshot 1`] = `
<div
  className="forecastText"
>
  <h2>
    Search for a city
  </h2>
</div>
`;
